source/AI.cpp

```diff
@@ -141,6 +143,36 @@ namespace {
 				bay.ship->SetCommands(Command::DEPLOY);
 	}
 
+	// Helper function for selecting the ships for formation commands.
+	vector<Ship *> GetShipsForFormationCommand(const PlayerInfo &player)
+	{
+		// Figure out what ships we are giving orders to.
+		vector<Ship *> targetShips;
```

Comment:
talking micro optimization but if we make this static we can avoid some memalloc calls

Reply:
also with a static we could pass a reference instead of copying the vector

Reply:
If we make this static, then we avoid some mem-alloc calls, but it also means that it uses some heap memory when we are not actively calling this function. This function is relatively rarely used compared to other functions (only when the player issues a formation command), so I prefer not to use the memory when it is not needed.
Also, this function is not alone, there are about 3 other functions here that have a similar way of working. (So if we fix this, then let's also addres the other functions that look very similar.)

Reply:
Makes sense